src/libponyc/expr/match.c

```diff
@@ -461,6 +461,7 @@ bool expr_case(pass_opt_t* opt, ast_t* ast)
       ast_error(opt->check.errors, pattern, "this pattern can never match");
       ast_error_continue(opt->check.errors, match_type, "match type: %s",
         ast_print_type(operand_type));
+      // TODO report unaliased type when body is consume !
```

Comment:
This looks like another case where the "deferred error context" discussed during this week's sync and summarised in #1972 could be useful.